cloudweatherreport/datastore.py

```diff
@@ -92,8 +87,7 @@ def lock(self, path=None, timeout=5*60):
         # consistency to ensure it will be immediately visible to others,
         # and their lock files to us.
         lock_filename = '.lock.{}'.format(uuid4())
-        lock_filepath = self._path(path, lock_filename)
-        self.write(lock_filepath, '')
+        self.write(lock_filename, '')
```

Comment:
This drops the value of `path` entirely and means that a created lock will never be visible, allowing all lock requests to succeed.  I think the real problem is with the double-concatenation of `self.prefix` in `self._path()` and that should be checked for and handled better there.